### PR TITLE
feat: add additional ci stages

### PR DESCRIPTION
**Pull request by jeff-mccoy**
Export `KubernetesObject` and `KubernetesListObject` as often-needed imports for [Pepr](https://pepr.dev/). 

New CI checks:
- dependabot
- codeql
- unit tests (node 18 + latest)
- commitlint PR title validation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,9 @@
     "prebuild": "rm -rf dist",
     "build": "tsc",
     "semantic-release": "semantic-release",
-    "test": "jest src --coverage"
+    "test": "jest src --coverage",
+    "format:check": "eslint src && prettier src --check",
+    "format:fix": "eslint src --fix && prettier src --write"
```
Comment (Noxsios) [inline package.json:13]: this won't cover github workflows / anything outside of `src`. Do you want to instead use `*.yaml|.yml|.js|.ts| ...`

File: .github/workflows/commitlint.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@5e21ff4d9bc1a8cf6de233a3057d20ec6b3fb69d # v3.8.1
+
+      - name: Install commitlint
+        run: npm install --save-dev @commitlint/{config-conventional,cli}
+
+      - name: Lint PR title
+        run: echo "${{ github.event.pull_request.title }}" | npx commitlint
```
Comment (Noxsios) [inline .github/workflows/commitlint.yml:27]: the error returned from this is a little vague, is there a way to make it more indicative of what the PR title format is?

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@cdcdbb579706841c47f7063dda365e292e5cad7a # v2.13.4
+        with:
+          languages: ${{ matrix.language }}
+
+      # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+      # If this step fails, then you should remove it and run the build manually (see below)
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@cdcdbb579706841c47f7063dda365e292e5cad7a # v2.13.4
```
Comment (Noxsios) [inline .github/workflows/codeql.yml:50]: is this needed for a pure `js` and `ts` project?

File: .github/workflows/dependency-review.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  contents: read
+
+jobs:
+  dependency-review:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout Repository'
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+      - name: 'Dependency Review'
+        uses: actions/dependency-review-action@6c5ccdad469c9f8a2996bfecaec55a631a347034 # v3.1.0
```
Comment (Noxsios) [inline .github/workflows/dependency-review.yml:22]:
```suggestion
name: Dependency Review
on: [pull_request]

permissions:
  contents: read

jobs:
  dependency-review:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout Repository
        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
      - name: Dependency Review
        uses: actions/dependency-review-action@6c5ccdad469c9f8a2996bfecaec55a631a347034 # v3.1.0
```

File: .github/workflows/codeql.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
```
Comment (Noxsios) [inline .github/workflows/codeql.yml:12]:
```suggestion
name: CodeQL
```

File: .github/workflows/node.js.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
+name: Node.js CI
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
```
Comment (Noxsios) [inline .github/workflows/node.js.yml:8]:
```suggestion

on:
  push:
    branches: [main]
  pull_request:
    branches: [main]

```

**Review by Noxsios — COMMENTED** [review]
some nitpicks about formatting

File: .github/workflows/commitlint.yml (hunk 2 above)
Comment (jeff-mccoy) [reply .github/workflows/commitlint.yml:27]: yeah that's the output of the tool--think we'll just we actually get a contrib guide

**Review by Noxsios — APPROVED** [review]

**Pull request merged**